Gems/Atom/RHI/Code/Tests/MultiDevicePipelineStateTests.cpp

```diff
@@ -0,0 +1,378 @@
+        RHI::PipelineStateDescriptorForDraw CreatePipelineStateDescriptor(uint32_t randomSeed)
+        {
+            RHI::PipelineStateDescriptorForDraw desc;
+            desc.m_inputStreamLayout.SetTopology(RHI::PrimitiveTopology::TriangleList);
+            desc.m_inputStreamLayout.Finalize();
+            desc.m_pipelineLayoutDescriptor = m_pipelineLayout;
+
+            ScrambleMemory(&desc.m_renderStates, sizeof(RHI::RenderStates), randomSeed++);
+
+            desc.m_renderStates.m_depthStencilState.m_depth.m_enable = true;
+            auto& renderAttachmentLayout = desc.m_renderAttachmentConfiguration.m_renderAttachmentLayout;
+            renderAttachmentLayout.m_attachmentCount = 2;
+            renderAttachmentLayout.m_attachmentFormats[0] = RHI::Format::R32_FLOAT;
+            renderAttachmentLayout.m_attachmentFormats[1] = RHI::Format::R8G8B8A8_SNORM;
+            renderAttachmentLayout.m_subpassCount = 1;
+            renderAttachmentLayout.m_subpassLayouts[0].m_rendertargetCount = 1;
+            renderAttachmentLayout.m_subpassLayouts[0].m_rendertargetDescriptors[0] =
+                RHI::RenderAttachmentDescriptor{ 1, RHI::InvalidRenderAttachmentIndex, RHI::AttachmentLoadStoreAction() };
+            renderAttachmentLayout.m_subpassLayouts[0].m_depthStencilDescriptor =
+                RHI::RenderAttachmentDescriptor{ 0, RHI::InvalidRenderAttachmentIndex, RHI::AttachmentLoadStoreAction() };
+            desc.m_renderAttachmentConfiguration.m_subpassIndex = 0;
+            return desc;
+        }
+
+        void ValidateCacheIntegrity([[maybe_unused]] RHI::Ptr<RHI::PipelineStateCache>& cache) const
+        {
+            EXPECT_TRUE(false);
+            /*
+            TODO: Once cache becomes multi-device, add that back in and also add MultiDevicePipelienStateTests as friend class to
+            PipelineStateCache
+             */
+            // cache->ValidateCacheIntegrity();
+        }
+
+    private:
+        void SetUp() override
+        {
+            MultiDeviceRHITestFixture::SetUp();
+
+            m_pipelineLayout = RHI::PipelineLayoutDescriptor::Create();
+            m_pipelineLayout->Finalize();
+        }
+
+        void TearDown() override
+        {
+            m_pipelineLayout = nullptr;
+            MultiDeviceRHITestFixture::TearDown();
+        }
+
+        RHI::Ptr<RHI::PipelineLayoutDescriptor> m_pipelineLayout;
+    };
+
+    TEST_F(MultiDevicePipelineStateTests, PipelineState_CreateEmpty_Test)
+    {
+        RHI::Ptr<RHI::MultiDevicePipelineState> empty = aznew RHI::MultiDevicePipelineState;
+        EXPECT_NE(empty.get(), nullptr);
+        EXPECT_FALSE(empty->IsInitialized());
+    }
+
+    TEST_F(MultiDevicePipelineStateTests, PipelineState_Init_Test)
+    {
+        RHI::Ptr<RHI::MultiDevicePipelineState> pipelineState = aznew RHI::MultiDevicePipelineState;
+        RHI::ResultCode resultCode = pipelineState->Init(DeviceMask, CreatePipelineStateDescriptor(0));
```

Comment:
where is DeviceMask defined?

Reply:
in `RHI/Code/Tests/Device.h` and it is set to `RHI::MultiDevice::AllDevices` and a device count of `8` is used.